lib/pipefy_message/consumer.rb

```diff
@@ -72,24 +72,36 @@ def process_message
 
         build_consumer_instance.poller do |payload, metadata|
           start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :millisecond)
-          logger.info({
-                        message_text: "Message received by poller to be processed by consumer",
-                        received_message: payload,
-                        metadata: metadata
-                      })
+
+          context = metadata["context"]
+          correlation_id = metadata["correlationId"]
+          event_id = metadata["eventId"]
+
+          logger.info(log_context({
+                                    message_text: "Message received by poller to be processed by consumer",
+                                    received_message: payload,
+                                    metadata: metadata
+                                  }, context, correlation_id, event_id))
 
           retry_count = metadata["ApproximateReceiveCount"].to_i - 1
           obj.perform(payload["Message"], { retry_count: retry_count })
 
           elapsed_time_ms = Process.clock_gettime(Process::CLOCK_MONOTONIC, :millisecond) - start
-          logger.info({
-                        duration_ms: elapsed_time_ms,
-                        message_text: "Message received by consumer poller, processed " \
-                                      "in #{elapsed_time_ms} milliseconds"
-                      })
+          logger.info(log_context({
+                                    duration_ms: elapsed_time_ms,
+                                    message_text: "Message received by consumer poller, processed " \
+                                                  "in #{elapsed_time_ms} milliseconds"
+                                  }, context, correlation_id, event_id))
         end
       rescue PipefyMessage::Providers::Errors::ResourceError => e
-        logger.error("Failed to process message, details #{e.inspect}")
+        context = "NO_CONTEXT_RETRIEVED" unless defined? context
+        correlation_id = "NO_CID_RETRIEVED" unless defined? correlation_id
+        event_id = "NO_EVENT_ID_RETRIEVED" unless defined? event_id
```

Comment:
I think this situation will not happen, given that in publisher.rb line 20 we always generate a UUID as the `event_id` and a message is published on SQS only by the SNS(who uses the same flow mentioned ☝🏾 ).

Reply:
I tried to keep this more general as I figured something could go wrong e.g. while polling the queue, before a message is received. There could also be a problem when trying to retrieve those values. I dunno, assuming that a variable is defined inside a rescue block looks like it could lead to an exception within an exception (probably an inception) haha